src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -263,11 +273,13 @@ function createEmscriptenModuleInstance(loadAssemblyUrls: string[], onReady: ()
       'number',
     ]);
 
-    mono_call_assembly_entry_point = Module.mono_call_assembly_entry_point;
```

Comment:
You can image instead that `Module.mono_call_assembly_entry_point` would just return a value, which could a `Task`, `int`, or `Task<int>`. That seems problematic because from the JS side we don't know the value's type - which leads back to us wanting to call the entry point ourselves.

Reply:
If Mono adds native support for async main, I think they should create a new entrypoint API called something like `mono_call_assembly_entry_point_async`, which returns a `Promise` that resolves with either `null`/`undefined` in the case of a `void` or `Task` main, or a `number` for a main of type `int`or `Task<int>`, or rejects with some representation of an exception for either sync or async failures.

I don't think there's any problem on the JS side with needing to know the type in advance.

Reply:
Its possible to do this more reliably by calling the 'mono_assembly_get_object' mono API function which returns an
Assembly object, then calling its EntryPoint property from c#. mono_assembly_get_object () is not currently exposed to js, but we can add it if needed.

Reply:
Thanks @vargaz - that's a great suggestion!

I realised we don't need `mono_assembly_get_object` either, as we can use `Assembly.Load(string)` on the .NET side to get the same result. If you think `mono_assembly_get_object` would be faster for any reason, then yes it would be great to have that exposed to JS. Would it be correct to expect that it wouldn't really be significantly faster, since `Assembly.Load(string)` is returning the `Assembly` instance already cached in memory anyway (since we already loaded it inside the bootup js)?

Reply:
mono_assembly_get_object () would be a bit faster. Will look into adding it.